src/_guides/language/language-tour.md

```diff
@@ -1042,6 +1042,26 @@ src="{{site.custom.dartpad.embed-dart-prefix}}?id=d988cfce0a54c6853799&horizonta
 </iframe>
 {% endcomment %}
 
+### Functional types
+
+Dart allows you to specify function types in places like variables and
```

Comment:
A statement like

> Dart allows you to specify function types in places like variables and fields.

only really makes sense under the implicit context of comparing the original (<1.24) anonymous function-type syntax with the new.

Dart has always had syntax for anonymous function-types, but the new syntax has no restrictions (as far as I can tell): you can use the new anonymous function-type syntax _anywhere_ a type annotation can be written. That is not the case with the original syntax which, in particular, cannot be used to annotate a global variable, local variable, or a class field.

IMHO, to be complete this new section should document both the new and the original syntax (in case the reader comes across the original syntax --- it will take some time before existing code is rewritten to use the new syntax, if at all).

How about opening this section something like this:

> ### Anonymous function-types
>
> In Dart, you can use an anonymous function-type as a type annotation. 
> Here are some examples:
> 
> [Example of, say, a callback function parameter using the original syntax, and a global variable using the new syntax, followed by a statement saying that both syntaxes will be explained next]
>
> You can write the anonymous type corresponding to any function.
> To do this, replace a function's name with `Function` in its declaration. For example,
> the type of `void foo() {...}` can be written as`void Function()`. ...
> This `Function`-keyword based syntax can be used anywhere a type annotation is permitted.
>
> In contrast, the original anonymous function-type syntax ...